File: Code Forces and Leetcode4/A_Anton_and_Letters.py
def count_distinct_letters(input_string):
    # Check if the input string is empty
    if input_string == "{}":
        return 0

    # Extract letters between curly braces
    letters = input_string[1:-1].split(", ")

    # Use a set to store distinct letters
    distinct_letters = set()

    # Iterate through letters and add to the set
    for letter in letters:
        distinct_letters.add(letter)

    # Return the number of distinct letters
    return len(distinct_letters)

# Read input
input_string = input().strip()

# Call the function and print the result
result = count_distinct_letters(input_string)
print(result)

File: Code Forces and Leetcode4/A_Anton_and_Letters.cpp
#include <iostream>
#include <algorithm>
#include <unordered_set>

using namespace std;

int countDistinctLetters(string& str) {
    unordered_set<char> distinctLetters;

    for (char ch : str) {
        if (isalpha(ch)) {  // Check if the character is a letter
            distinctLetters.insert(ch);
        }
    }

    return distinctLetters.size();
}

int main() {
    string str = "{a, a, a, a}";

    // Extracting the substring between curly brackets
    size_t start = str.find('{');
    size_t end = str.find('}');
    string substring = str.substr(start + 1, end - start - 1);

    int result = countDistinctLetters(substring);

    cout << result << endl;

    return 0;
}
